### ISSUE TITLE
Regex does not support unicode category "Currency Symbol"

### ISSUE THREAD

**Issue by wzzzzd** [maciejhirsz/logos#359]
I'm writing a lexer for Cypher language. As specified in its [ANTLR grammar](https://s3.amazonaws.com/artifacts.opencypher.org/M23/Cypher.g4), I define the token for unescaped identifiers as follows:
```rust
#[Logos, Debug]
pub enum Token {
    #[regex(r"[\p{ID_Start}\p{Pc}][\p{ID_Continue}\p{Sc}]*")]
    UnescapedSymbolicName,
    
    ...
}
```
It fails to compile due to regex parse error:
```bash
error: regex parse error:
           [\p{ID_Start}\p{Pc}][\p{ID_Continue}\p{Sc}]*
                                               ^^^^^^
       error: Unicode property not found
   --> src/token.rs:236:13
    |
236 |     #[regex(r"[\p{ID_Start}\p{Pc}][\p{ID_Continue}\p{Sc}]*")]
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

It seems the problem lies in the dependency `regex-syntax = 0.6`. I try to parse `[\p{Sc}]` with it:  
```rust
#[test]
fn test() {
    let mut parser = regex_syntax::Parser::new();
    let hir = parser.parse(r"\p{Sc}");
    println!("{:?}", hir);
}
```
And It raises the same error
```
Err(Translate(Error { kind: UnicodePropertyNotFound, pattern: "\\p{Sc}", span: Span(Position(o: 0, l: 1, c: 1), Position(o: 6, l: 1, c: 7)) }))
```

The problem might be solved by updating `regex-syntax`. For example, with `regex-syntax = 0.8.2`, the test above succeeds:
```
Ok(Class({'$'..='$', '¢'..='¥', '֏'..='֏', '؋'..='؋', '߾'..='߿', '৲'..='৳', '৻'..='৻', '૱'..='૱', '௹'..='௹', '฿'..='฿', '៛'..='៛', '₠'..='⃀', '꠸'..='꠸', '﷼'..='﷼', '﹩'..='﹩', '＄'..='＄', '￠'..='￡', '￥'..='￦', '𑿝'..='𑿠', '𞋿'..='𞋿', '𞲰'..='𞲰'}))
```

**Comment by jeertmans** [comment]
Hello, #320 already solved this issue by upgrading the regex-syntax dependency. The PR is stalled because I am waiting some reply from the author of this crate, as I cannot publish new releases on my own :/

**Comment by wzzzzd** [comment]
> Hello, #320 already solved this issue by upgrading the regex-syntax dependency. The PR is stalled because I am waiting some reply from the author of this crate, as I cannot publish new releases on my own :/

Nice to hear that. Hope the PR can be merged soon :)

**Comment by jeertmans** [comment]
@wzzzzd the issue is not the PR being merged, because I can do that anytime I want. But I cannot publish a new release :/

**Comment by jeertmans** [comment]
Fixed with 0.14.
